platform/o.n.bootstrap/src/org/netbeans/StandardModuleData.java

```diff
@@ -78,7 +80,12 @@ public StandardModuleData(Manifest mf, StandardModule forModule) throws InvalidE
         if (classPath != null) {
             StringTokenizer tok = new StringTokenizer(classPath);
             while (tok.hasMoreTokens()) {
-                String ext = tok.nextToken().replace("%20", " "); // NOI18N
+                String ext;
+                try {
+                    ext = URLDecoder.decode(tok.nextToken(), "UTF-8");
+                } catch (UnsupportedEncodingException ex) {
+                    throw new IllegalStateException(ex);
+                }
                 File extfile;
                 if (ext.equals("${java.home}/lib/ext/jfxrt.jar")) { // NOI18N
```

Comment:
It seems we do not really require that $ sign.
Also there are some JDK 1.7 checks which could be removed as well.

Reply:
Yes, the `${java.home}` string is totally artificial. I have chosen it as it had no meaning for `javac` (in JDK8, who could expect JDK11 starts to recognize URLs in `Class-Path` attribute!).

There already was some URI-like processing (replacing `%20` by ` `) and using `URLDecoder` (not available on JDK7 at the time of writing this code) can be seen as a "proper fix" of processing the URI component.

Right, the JDK7 specific code can be removed.

Reply:
FWIW, you can probably use `java.nio.charset.StandardCharsets.UTF_8` instead of `"UTF-8"`, and avoid the need to handle the `UnsupportedEncodingException`.

Reply:
Alas, I cannot. That method has been added in JDK10! While it seems available in the editor, when compiling with the new `--release 8` the compilation fails. I tried and had to revert back to specifying the encoding as string and catching the exception.